### ISSUE TITLE
Add option to grid_coordinates to disable calling meshgrid

### ISSUE THREAD

**Issue by leouieda** [fatiando/verde#382]
**Description of the desired feature:**

Currently, `verde.grid_coordinates` always calls `np.meshgrid` to produce 2D coordinate arrays. However, when building `xarray` objects we usually have to undo. That's a bit wasteful can creates problems for very large grids. To counteract this, I propose adding a new `meshgrid=True` option to `grid_coordinates`. By default, the behaviour is unchanged. But passing `meshgrid=False` will return the 1D coordinate arrays instead.

**Are you willing to help implement and maintain this feature?** 

Always but happy for someone else to do this 🙂

**Comment by santisoler** [comment]
I agree with this! We would need to think what to do with the `extra_coords`. Any thoughts?

**Comment by leouieda** [comment]
Good point! It almost doesn't make sense to use `extra_coords` and `meshgrid=False` together since the whole point of `extra_coords` is to get a meshgrid with constant values. So maybe we just return it as is (i.e. a single value)?

**Comment by santisoler** [comment]
I agree that if we want to avoid allocating large coordinate arrays it doesn't make sense to generate huge `extra_coords` arrays with constant values. But it doesn't make any sense either to pass a single float (or list of floats) to `extra_coords` and get the same ones as output. What if we just avoid the trouble by ignoring `extra_coords` if we pass `meshgrid=False`?

**Comment by leouieda** [comment]
Sounds OK to me. Probably doesn’t matter too much which way we go since using both options together is extremely unlikely (famous last words 🥲).
